src/lib/y2network/interface_config_builders/bond.rb

```diff
@@ -0,0 +1,56 @@
+require "yast"
+require "y2network/config"
+require "y2network/interface_config_builder"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    class Bond < InterfaceConfigBuilder
+
+      include Yast::Logger
+
+      def initialize
+        super(type: "bond")
+      end
+
+      # @return [Array<Interface>] list of interfaces usable in bridge_iface
+      def select_bondable
+        interfaces.all.select { |i| bondable?(i) }
+      end
+
+
+    private
+
+      def interfaces
+        Config.find(:yast).interfaces
+      end
+
+      # Checks whether an interface can be enslaved in particular bond interface
+      #
+      # @param iface [Interface] an interface to be validated as bond_iface slave
+      # TODO: Check for valid configurations. E.g. bond device over vlan
+      # is nonsense and is not supported by netconfig.
+      # Also devices enslaved in a bridge should be excluded too.
+      def bondable?(iface)
+        Yast.import "Arch"
+        Yast.include self, "network/lan/s390.rb"
```

Comment:
as already written before including whole network include hell to new API is bad idea as it horrible mess object. Cleaner solution is to have private class which include it and call it from there.

Reply:
yes it is already on my todo list - just wanted to move it to final place before doing so ;-)